### ISSUE TITLE
Date Format Issue

### ISSUE THREAD

**Issue by hstoney** [LGoodDatePicker/LGoodDatePicker#76]
Attempting to move to version 10.4.1 my dates using a Day Name format all appeared as invalid (red). It is OK with version 10.3.1.
I changed to passing a date formatter to setFormatForDatesCommonEra but tah did not resolve the issue.

Here is my code that fails ( with the original line commetted before trying a date formatter:

datePicker = new DatePicker((DatePickerSettings) null);
DatePickerSettings dateSettingsPgmDate = new DatePickerSettings();
dateSettingsPgmDate.setAllowEmptyDates(true);
//		dateSettingsPgmDate.setFormatForDatesCommonEra("EEE, dd MMM yyyy");
DateTimeFormatter dateFormatter =  DateTimeFormatter.ofPattern("EEE, dd MMM yyyy");
dateSettingsPgmDate.setFormatForDatesCommonEra(dateFormatter);
datePicker = new DatePicker(dateSettingsPgmDate);
datePicker.getComponentDateTextField().setEditable(false);
datePicker.getComponentDateTextField().setToolTipText("The earliest date for booking to display.");
datePicker.getComponentToggleCalendarButton().setToolTipText("The earliest date for booking to display.");
datePicker.getComponentToggleCalendarButton().setText("+");
datePicker.getComponentToggleCalendarButton().setFont(new Font("Arial", Font.PLAIN, 12));
datePicker.getComponentDateTextField().setHorizontalAlignment(SwingConstants.LEFT);
datePicker.getComponentDateTextField().setFont(new Font("Arial", Font.PLAIN, 12));
datePicker.getComponentDateTextField().setDisabledTextColor(Color.BLACK);
datePicker.setBounds(115, 50, 160, 20);
datePicker.setDateToToday();
datePicker.addDateChangeListener(new DatePickerChangeListener("datePicker"));
getContentPane().add(datePicker);

I suspect the fix for issue #60 may be the cause.

I will stick with version 10.3.1 as it is a good fit for my needs and I have found your date picker good to work with.

**Comment by stromjust** [comment]
Same issue.

**Comment by WiseEagleOwl** [comment]
@hstoney @stromjust 
This issue should be solved with the current *master*. The fixes were merged with PR #80 and PR #81.
In 10.4.1 there was a regression where formatter strings with `YearOfEra` (`yyyy` pattern) instead of `Year` (`uuuu` pattern) were not accepted.
In addition it is now possible to use case sensitive `DateTimeFormatter` which was not possible in 10.3.1.

The code given above uses both (case sensitive formatter and `YearOfEra` pattern):
```Java
DateTimeFormatter dateFormatter = DateTimeFormatter.ofPattern("EEE, dd MMM yyyy");
```
If you need to stick to an already released version using 10.3.1 with the following modification should be possible:
```Java
DateTimeFormatter dateFormatter = new DateTimeFormatterBuilder().parseCaseInsensitive()
.appendPattern("EEE, dd MMM yyyy").toFormatter();
```
Feel free to leave a comment if this hasn't solved the issue for you.
